patchset/0030-lavc-vaapi_hevc-fill-rext-luma-chroma-offset-in-the-.patch

```diff
@@ -0,0 +1,107 @@
+From 8fe75707f3142220f02720b88144f50a044c4362 Mon Sep 17 00:00:00 2001
+From: Xu Guangxin <guangxin.xu@intel.com>
+Date: Thu, 12 Mar 2020 12:04:53 -0400
+Subject: [PATCH] lavc/vaapi_hevc: fill rext luma/chroma offset in the right
+ way
+
+For range extension, the luma/chroma offset is larger than 8 bits, we need fill the 16 bits version
+---
+ libavcodec/vaapi_hevc.c | 48 +++++++++++++++++++++++++++--------------
+ 1 file changed, 32 insertions(+), 16 deletions(-)
+
+diff --git a/libavcodec/vaapi_hevc.c b/libavcodec/vaapi_hevc.c
+index d832b8cb0d..e1c4f26741 100644
+--- a/libavcodec/vaapi_hevc.c
++++ b/libavcodec/vaapi_hevc.c
+@@ -322,10 +322,17 @@ fail:
+     return ret;
+ }
+ 
+-static void fill_pred_weight_table(const HEVCContext *h,
++static void fill_pred_weight_table(const AVCodecContext *avctx,
++                                   const HEVCContext *h,
```

Comment:
Could use h->ps->sps to fetch the sps information, so maybe no need to add another entry (avctx) for this function.

Reply:
we need profile informaiton, not sps.

Reply:
how about h->ps->sps->sps_range_extension_flag?

Reply:
sps_range_extension_flag just means sps_range_extension( ) existed or not. It's not a indicate for range extension stream.
ie sps_range_extension( ) may not existed for a range extension bitstream